openhantek/src/hantekdso/hantekdsocontrol.cpp

```diff
@@ -271,14 +275,14 @@ void HantekDsoControl::convertRawDataToSamples(const std::vector<unsigned char>
                     ((unsigned short int)rawData[totalSampleCount + bufferPosition - extraBitsPosition] << shift) &
                     extraBitsMask;
 
-                result.data[channel][pos] = ((double)(low + high) / limit - offset) * gainStep;
+                result.data[channel][pos] = ((double)(low + high) / limit - offset) * gainStep * probeGain;
```

Comment:
HantekDsoControl should output the pure samples, only adapted to the devices abilities (10bit, limits, offsets). Your signal modification is more a post processing change and belongs into the `post` directory as a separate class that registers to the PostProcessing class. See the math channel generator class in the same directory as an example.

Reply:
My idea was this: the probe attenuation gives in output a voltage that is actually not the real one, but one attenutate, so it would make sense to process the raw voltage to adjust for that.